bpf/proxy/syncer.go

```diff
@@ -221,9 +221,62 @@ func (s *Syncer) loadOrigs() error {
 	return nil
 }
 
-func (s *Syncer) startupSync(state DPSyncerState) error {
+type syncRef struct {
+	svc  k8sp.ServicePortName
+	info k8sp.ServicePort
+}
+
+func (s *Syncer) svcMapToIPPortProtoMap(svcs k8sp.ServiceMap) map[nat.FrontendKey]syncRef {
```

Comment:
I think it'd help to add godoc to these methods that explains their higher-level purpose.  For example, this one takes the kubernetes service representation and makes an index so we can cross reference with the values we learn from the dataplane.